### PR TITLE
Rl v3 hanging issue fix

### PR DESCRIPTION
**Pull request by ysqyang**
# Description

1. Fixed hanging issue in distributed training;
2. Updated workflows and CLI; 
3. Moved CIM example back to example folder

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/rl_v3/distributed/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from .remote_obj import Client, RemoteObj
```
Comment (lihuoran) [inline maro/rl_v3/distributed/__init__.py:1]: Add `Client` and `RemoteObj` into `__all__`

File: maro/rl_v3/training/algorithms/maddpg.py [hunk 2]
```diff
@@ -363,26 +367,24 @@ def record(self, exp_element: ExpElement) -> None:
...
-                "shared_critic": False,
-                "create_actor": False,
-            }
-            return DiscreteMADDPGOps(**params)
+            return DiscreteMADDPGOps(
+                get_policy_func=None,
+                policy_idx=-1,
+                shared_critic=False,
+                create_actor=False,
+                **self._ops_params
```
Comment (lihuoran) [inline maro/rl_v3/training/algorithms/maddpg.py:377]: `self._ops_params` should be put at the first place, and let the following parameters to overwrite. For example, `shared_critic`.

File: maro/rl_v3/training/ops_creator.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+from .trainer import AbsTrainer
+
+
+class OpsCreator(object):
+    def __init__(
+        self,
+        policy_creator: Dict[str, Callable[[str], RLPolicy]],
+        trainer_creator: Dict[str, Callable[[str], AbsTrainer]]
+    ):
```
Comment (lihuoran) [inline maro/rl_v3/training/ops_creator.py:17]: More type hints?

File: maro/rl_v3/training/trainer_manager.py [hunk 4]
```diff
@@ -73,32 +73,27 @@ def __init__(
...
+        asyncio.run(self._gather("train_step"))
 
     def get_policy_states(self) -> Dict[str, Dict[str, object]]:
-        policy_state_list = asyncio.run(self._get_policy_states())
+        policy_state_list = asyncio.run(self._gather("get_policy_state"))
         return dict((trainer.name, policy_state) for trainer, policy_state in zip(self._trainers, policy_state_list))
 
-    async def _get_policy_states(self):
-        return await asyncio.gather(*[trainer.get_policy_state() for trainer in self._trainers])
+    async def _gather(self, coroutine_name):
```
Comment (lihuoran) [inline maro/rl_v3/training/trainer_manager.py:95]: type hint

File: maro/rl_v3/rollout/__init__.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from .env_sampler import AbsAgentWrapper, AbsEnvSampler, CacheElement, ExpElement, SimpleAgentWrapper
+
+__all__ = [
+    "AbsAgentWrapper", "AbsEnvSampler", "CacheElement", "ExpElement", "SimpleAgentWrapper",
+    "AbsTrainerManager", "SimpleTrainerManager"
```
[the remark this answers is not in the document]
Comment (ysqyang) [reply maro/rl_v3/rollout/__init__.py:5]: This folder should not be in the PR

**Pull request merged**